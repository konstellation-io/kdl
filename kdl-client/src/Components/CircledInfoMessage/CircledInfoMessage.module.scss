@import 'Styles/colors';
@import 'Styles/variables';
@import 'Styles/mixins';

.circledMessage {
  @include font-small;
  padding-left: 2 * $grid-unit;
  line-height: 3 * $grid-unit;
  margin: 0 $grid-unit / 2;

  &.success {
    @include circle-before(palette(success));
    color: palette(success);
  }

  &.error {
    @include circle-before(palette(alert));
    color: palette(alert);
  }
}
