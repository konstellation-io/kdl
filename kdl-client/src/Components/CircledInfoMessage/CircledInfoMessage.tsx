import React from 'react';

import styles from './CircledInfoMessage.module.scss';
import cx from 'classnames';

export enum CircledInfoMessageTypes {
  SUCCESS = 'success',
  ERROR = 'error',
}

type Props = {
  type: CircledInfoMessageTypes;
  text?: string;
  className?: string;
};

const CircledInfoMessage: React.FC<Props> = ({
  type = CircledInfoMessageTypes.SUCCESS,
  text = '',
  children,
}) => {
  return (
    <div className={cx(styles.circledMessage, styles[type])}>
      {text}
      {children}
    </div>
  );
};

export default CircledInfoMessage;
