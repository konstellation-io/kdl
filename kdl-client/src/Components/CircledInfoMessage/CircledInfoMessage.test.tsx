import CircledInfoMessage, {
  CircledInfoMessageTypes,
} from './CircledInfoMessage';

import React from 'react';
import { shallow } from 'enzyme';

const component = shallow(
  <CircledInfoMessage type={CircledInfoMessageTypes.SUCCESS} text="some text">
    <div id="children">Children</div>
  </CircledInfoMessage>
);

describe('CircledInfoMessage component', () => {
  test('Component match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('show right texts', () => {
    expect(component.contains('some text')).toBeTruthy();
    expect(component.contains('Children')).toBeTruthy();
  });

  it('contains right classes', () => {
    expect(component.find('.success').isEmptyRender()).toBeFalsy();
    expect(component.find('.error').isEmptyRender()).toBeTruthy();

    component.setProps({ type: CircledInfoMessageTypes.ERROR });

    expect(component.find('.success').isEmptyRender()).toBeTruthy();
    expect(component.find('.error').isEmptyRender()).toBeFalsy();
  });
});
