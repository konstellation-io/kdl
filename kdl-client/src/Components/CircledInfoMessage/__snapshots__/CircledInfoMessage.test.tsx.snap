// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircledInfoMessage component Component match snapshot 1`] = `
<div
  className="circledMessage success"
>
  some text
  <div
    id="children"
  >
    Children
  </div>
</div>
`;
